### PR TITLE
[BUG] Fix camera view calculation and tidy up system.

### PR DESCRIPTION
**Pull request by DelinWorks**
Refer to #791
The camera features that I added were bandaids over bandaids so I decided to rewrite it and it's significantly better now!

### CODE REVIEW

File: core/2d/CCCamera.cpp [hunk 1]
```diff
@@ -42,7 +42,37 @@ Viewport Camera::_defaultViewport;
 Camera* Camera::create()
 {
     Camera* camera = new Camera();
-    camera->initDefault();
+
+    auto& size = Director::getInstance()->getWinSize();
```
Comment (halx99) [inline core/2d/CCCamera.cpp:46]:
```suggestion
    auto& size = _director->getWinSize();
```

File: core/2d/CCCamera.cpp [hunk 2]
```diff
@@ -42,7 +42,37 @@ Viewport Camera::_defaultViewport;
...
+    switch (Director::getInstance()->getProjection())
```
Comment (halx99) [inline core/2d/CCCamera.cpp:47]:
```suggestion
    switch (_director->getProjection())
```

File: core/2d/CCCamera.cpp [hunk 3]
```diff
@@ -195,56 +221,33 @@ void Camera::setAdditionalProjection(const Mat4& mat)
...
-    {
-        applyCustomProperties();
-        return true;
-    }
-    _isCameraInitialized = true;
-
     auto& size = _director->getWinSize();
     // create default camera
-    switch (_projectionType)
+    switch (Director::getInstance()->getProjection())
```
Comment (halx99) [inline core/2d/CCCamera.cpp:228]:
```suggestion
    switch (_director->getProjection())
```

File: core/2d/CCCamera.cpp [hunk 4]
```diff
@@ -405,7 +408,7 @@ void Camera::setZoom(float factor)
 
 void Camera::applyZoom()
 {
-    switch (_projectionType)
+    switch (Director::getInstance()->getProjection())
```
Comment (halx99) [inline core/2d/CCCamera.cpp:411]:
```suggestion
    switch (_director->getProjection())
```

**Review by halx99 — CHANGES_REQUESTED** [review]

File: core/2d/CCCamera.cpp (hunk 1 above)
Comment (DelinWorks) [reply core/2d/CCCamera.cpp:46]: `_director` cannot be used in a static function.

File: core/2d/CCCamera.cpp [hunk 5]
```diff
@@ -42,7 +42,37 @@ Viewport Camera::_defaultViewport;
...
+                up(0.0f, 1.0f, 0.0f);
+            camera->setPosition3D(eye);
+            camera->lookAt(center, up);
+            break;
+        }
+    }
+
+    camera->updateTransform();
     camera->autorelease();
     camera->setDepth(0);
```
Comment (halx99) [inline core/2d/CCCamera.cpp:77]: The code block still can wraped as a member function of Camera.

**Review by halx99 — APPROVED** [review]

File: core/2d/CCCamera.cpp [hunk 6]
```diff
@@ -42,7 +42,7 @@ Viewport Camera::_defaultViewport;
 Camera* Camera::create()
 {
     Camera* camera = new Camera();
-    camera->initDefault();
+    camera->initCamera();
```
Comment (halx99) [inline core/2d/CCCamera.cpp:45]: Can be still name `initDedault`?

File: core/2d/CCCamera.cpp [hunk 7]
```diff
@@ -42,7 +42,7 @@ Viewport Camera::_defaultViewport;
...
     camera->autorelease();
     camera->setDepth(0);
```
Comment (halx99) [inline core/2d/CCCamera.cpp:47]: Can be move to `initDefault`?

File: core/2d/CCCamera.cpp [hunk 8]
```diff
@@ -53,11 +53,9 @@ Camera* Camera::createPerspective(float fieldOfView, float aspectRatio, float ne
 {
     auto ret = new Camera();
     ret->_fieldOfView = fieldOfView;
```
Comment (halx99) [inline core/2d/CCCamera.cpp:55]: Does the 3 assign statements can be delete?

File: core/2d/CCCamera.cpp [hunk 9]
```diff
@@ -70,7 +68,6 @@ Camera* Camera::createOrthographic(float zoomX, float zoomY, float nearPlane, fl
     ret->_nearPlane = nearPlane;
     ret->_farPlane  = farPlane;
```
Comment (halx99) [inline core/2d/CCCamera.cpp:69]: Does the 4 assign statements can be delete?

File: core/2d/CCCamera.cpp (hunk 6 above)
Comment (DelinWorks) [reply core/2d/CCCamera.cpp:45]: it's gonna be misleading if `initDefault` is used, previous code bases will think it's an old API function and will lead to bugs.

Comment (DelinWorks) [reply core/2d/CCCamera.cpp]: Actually, it needs to be `init` only overriding the node's `init` function, that way the code is cleaner

**Comment by DelinWorks** [comment]
The ci error is caused by `lua auto generated bindings` I don't think they need to be fixed because they're auto, that's what I think

**Comment by halx99** [comment]
These changes can't fix #791, I fix with the commit https://github.com/axys1/axys/commit/3cf19787ce1bb4945e42c73e013c735c7a046bd2, and I rollback default camera init from `init` to `initDefault`.

**Comment by halx99** [comment]
> These changes can't fix #791, I fix with the commit [3cf1978](https://github.com/axys1/axys/commit/3cf19787ce1bb4945e42c73e013c735c7a046bd2), and I rollback default camera init from `init` to `initDefault`.

I'm not sure does this will break your features, I think you can check it

**Pull request merged**